### ISSUE TITLE
Missing test and example for chol input to MvNormal

### ISSUE THREAD

**Issue by twiecki** [pymc-devs/pymc#2019]
I tried using the new cholesky input to MvNormal but it didn't seem to work (ADVI gave nans) and there were no example docs (should have a NB) and also couldn't see tests.

CC @aseyboldt

**Comment by aseyboldt** [comment]
I'll have a look tonight and add an example to the docstring. The missing tests are strange, i clearly remember seeing them in the pr, but can't find them right now.
I planned to cover it in the lkj notebook (I'm working on that). Do you think we need a dedicated notebook for the mv normal? And can you be more specific about what doesn't work?

On April 11, 2017 5:08:35 PM CEST, Thomas Wiecki <notifications@github.com> wrote:
>I tried using the new cholesky input to MvNormal but it didn't seem to
>work (ADVI gave nans) and there were no example docs (should have a NB)
>and also couldn't see tests.
>
>CC @aseyboldt
>
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/pymc-devs/pymc3/issues/2019

-Adrian

**Comment by twiecki** [comment]
@aseyboldt LKJ notebook will be perfect. I passed in an upper-triangular matrix to chol, is that wrong? it works with `chol.dot(chol.T)`

**Comment by aseyboldt** [comment]
You need a lower triangular. But I guess we should add a keyword for that. And maybe change the default before the release. Scipy uses upper as default, right?

On April 11, 2017 5:38:06 PM CEST, Thomas Wiecki <notifications@github.com> wrote:
>@aseyboldt LKJ notebook will be perfect. I passed in an
>upper-triangular matrix to chol, is that wrong? it works with
>`chol.dot(chol.T)`
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/pymc-devs/pymc3/issues/2019#issuecomment-293303436

-Adrian

**Comment by twiecki** [comment]
We just passed in whatever was returned by expand_packed_chol.

**Comment by aseyboldt** [comment]
This works (and is more than twice as fast as stan)
```python
N = 50
cov = np.eye(N)
for i in range(len(cov)):
    if i > 0:
        cov[i, i-1] = 0.2
    if i + 1 < len(cov):
        cov[i, i+1] = 0.2

vals = np.random.multivariate_normal(np.zeros(N), cov=cov, size=500)

with pm.Model() as model:
    sd_dist = pm.HalfCauchy.dist(beta=2.5)
    chol_packed = pm.LKJCholeskyCov('chol', n=N, eta=2, sd_dist=sd_dist)
    chol = pm.expand_packed_triangular(N, chol_packed, lower=True)
    mu = np.zeros(N)
    pm.MvNormal('vals', mu=mu, chol=chol, observed=vals)
```

**Comment by aseyboldt** [comment]
fixed by #2028
